guide/interactions/registering-slash-commands.md

```diff
@@ -0,0 +1,124 @@
+# Registering slash commands
+
+Discord provides users the option to create client-integrated slash commands. In this section you will be learning how to register these commands using discord.js!
+
+::: tip
+If you already have slash commands set-up for your application and want to learn to respond to them, refer to [the following page](/interactions/replying-to-slash-commands/).
+:::
+
+
+## Global commands
+
+First up, we'll introduce you to global application commands, these type of commands will be available on every guild your application is authorized on with the `applications.commands` oauth scope and are also available in its DMs.
+
+::: warning
+Global commands are cached for one hour. That means that new global commands will fan out slowly across all guilds, and will be guaranteed to be updated in an hour. 
+:::
+
+::: tip
+Guild commands update instantly. We recommend you use guild commands for quick testing, and global commands when they're ready for public use.
+:::
+
+So, to register a global command we will be passing an `ApplicationCommandData` object to the `ApplicationCommandManager.create()` method as follows:
+
+```js
+client.once('ready', async () => {
+	const commanddata = {
+		name: 'ping',
+		description: 'Replies with Pong!'
+	};
+
+	const command = await client.application?.commands.create(commanddata);
```

Comment:
```suggestion
	const command = await client.application.commands.create(commanddata);
```

Reply:
The property is nullable, so the optional chaining operator here is a good practice.

Reply:
I agree that it's not needed, `client.application` is always set after the ready event was emitted. Also, if there's no error and that call just gets ignored, people will wait an hour for the command to appear only to realize that it hasn't been created in the first place.

Reply:
I agree with that point, I did however also use the operator on the guild commands section, get() can return undefined as well, so I'm unsure how the guide approaches optional chaining overall. Should it stay for possibly undefined props but not for possibly null props, should it be gone entirely?, I think this requires some consistency, but I don't know which fits best. 
TypeScript will definitely enforce the operator anyway so again not sure what applies best